snowpack/src/scan-imports.ts

```diff
@@ -273,7 +264,8 @@ export async function scanImportsFromFiles(
   config: SnowpackConfig,
 ): Promise<InstallTarget[]> {
   return loadedFiles
-    .map(parseFileForInstallTargets)
+    .filter((sourceFile) => !Buffer.isBuffer(sourceFile.contents)) // filter out binary files from import scanning
```

Comment:
if we already filter by `.js`, `.ts`, etc, above should we also filter here? Or, should this really be a warning of "hold up, why is there binary data in a JS file?"

Reply:
That’s kind of the thing—this install step runs in both `build` and `dev` and the reading of those files happens outside this function sometimes. So in order to not have this, we’d have to make an even more sweeping change.

[Example](https://github.com/pikapkg/snowpack/blob/master/snowpack/src/commands/install.ts#L449):

```ts
  if (scannedFiles) {
    installTargets.push(...(await scanImportsFromFiles(scannedFiles, config)));
  } else {
    installTargets.push(...(await scanImports(cwd, config)));
  }
```

We have to handle many different places within Snowpack reading files, which means multiple code paths where binary files could slip through. I could change all that, but that would have been a bigger PR.

Reply:
got it, thanks for explaining 👍